server/index.js

```diff
@@ -32,30 +32,42 @@ app.get('/api/nearby/:carousel_id', (req, res) => {
     .catch(err => res.status(400).json(err));
 });
 
-app.put('/api/nearby/:carousel_id', (req, res) => {
-  db.addFavorite(req.params.carousel_id, req.query.restaurantId, req.query.increment)
-    .then(updated => res.status(202).send(updated.carousel))
+app.put('/api/nearby/', (req, res) => {
+  const { id, carousel } = req.body;
+  db.updateCarouselById(id, carousel)
+    .then((result) => {
+      res.status(200).send(result);
+    })
     .catch(err => res.status(400).json(err));
 });
 
-app.put('/api/nearby/:carousel_id', (req, res) => {
+app.put('/api/nearby/favorite/:carousel_id', (req, res) => {
+  console.log('faved!')
   db.addFavorite(req.params.carousel_id, req.query.restaurantId, req.query.increment)
     .then(updated => res.status(202).send(updated.carousel))
     .catch(err => res.status(400).json(err));
 });
 
-app.put('/api/nearbyUpdate/', (req, res) => {
-  const carouselUpdate = req.body;
-  const { id, carousel } = carouselUpdate;
-  db.updateCarouselById(id, carousel)
-    .then((result) => {
-      res.status(200).send(result);
-    })
-    .catch(err => res.status(400).json(err));
-});
-
 app.delete('/api/nearby/:carousel_id', (req, res) => {
-  db.deleteCarouselById(req.params.carousel_id)
-    .then(result => res.status(200).send(result))
-    .catch(err => res.status(400).send(err));
+  let targetCarousel;
+  db.findCarousel(req.params.carousel_id) // find the carousel so we can return it on deletion
+    .then((data) => {
+      if (data[0].carousel.length === 0) {
+        throw Error('Carousel not found');
+      } else {
+        targetCarousel = data;
+      }
+    })
+    .then(() => {
+      db.deleteCarouselById(req.params.carousel_id)
+        .then(() => {
```

Comment:
I think it's generally best to avoid nesting `then()`s like this, since it makes the code harder to think about. It would be nice if this `then()` were at the same level as the previous one.

Reply:
Ah thanks, it was an unexpected challenge to denest, but not too bad.